### ISSUE TITLE
Barcode scanner couldn't recognize EAN 8 and EAN 13 in Xamarin.UWP

### ISSUE THREAD

**Issue by huangjinshe** [Redth/ZXing.Net.Mobile#351]
I'm using UWP in Xamarin.Forms.

My Barcode scanner could regognize QR code but can't recognize EAN 8 and EAN 13.  what I missed in here?

Here is my code:

```
btnBarcode.Clicked += (sender, args) =>
            {
                Task.Run(async () =>
                {

                    await Task.Delay(100);
                    Xamarin.Forms.Device.BeginInvokeOnMainThread(async () =>
                    {

                        var options = new ZXing.Mobile.MobileBarcodeScanningOptions();
                        options.PossibleFormats = new List<ZXing.BarcodeFormat>() {ZXing.BarcodeFormat.EAN_8, ZXing.BarcodeFormat.EAN_13
};


                        var scanner = new ZXing.Mobile.MobileBarcodeScanner();

                        var result = await scanner.Scan(options);

                        if (result != null)
                        {
                            await DisplayAlert("barcode", result.Text, "cancel");
                        }


                    });
                });
            };
```

I also upload the sample file to:https://drive.google.com/file/d/0Bw0uognK2ErvYmxjanJieWxhUG8/view?usp=sharing

**Comment by huangjinshe** [comment]
I found the problem,  if you want to read EAN 8 and EAN 13, you need to rotate your label 90*.  I still not sure it's a bug or not, waiting for the owner confirm

**Comment by redwolf2** [comment]
Activate "TryHarder". If i remember correctly, it will rotate the image three times and try every direction.

**Comment by huangjinshe** [comment]
@redwolf2 I hope you could try it use current stable version. I remember I've already tried `TryHarder`.  but only could  recognize when rotate the phone to 90*. I hope let it support 0* rotate state could let it recognize.

**Comment by redwolf2** [comment]
https://github.com/Redth/ZXing.Net.Mobile/blob/master/Source/ZXing.Net.Mobile.WindowsUniversal/ZXingScannerControl.xaml.cs

This seem to be the related class

The relevant part is this:

`try
                {

                    // Get preview 
                    var frame = await mediaCapture.GetPreviewFrameAsync(destFrame);

                    // Create our luminance source
                    luminanceSource = new SoftwareBitmapLuminanceSource(frame.SoftwareBitmap);

                } catch (Exception ex)
                {
                    System.Diagnostics.Debug.WriteLine("GetPreviewFrame Failed: {0}", ex);
                }

                ZXing.Result result = null;

                try
                {
                    // Try decoding the image
                    if (luminanceSource != null)
                        result = zxing.Decode(luminanceSource);
                }
                catch (Exception ex)
                {
                    
}`

At this position you will have to modify the code, rotate the frame three times and run it through the zxing.Decode method until you get a result.

**Comment by huangjinshe** [comment]
Let's just don't say about Rotate 3 times.  I just want EAN code code recognize without rotate my phone 90 degrees. Then I even don't need to use `TryHarder`, or rotate 3 times.  Could you just tell me is that a UWP problem or the core?

**Comment by Redth** [comment]
This should be fixed in master now.  Basically the preview and record buffers weren't being properly when rotation occured.  This will be included in the 2.1.0 release.
